datadog_checks_base/tests/test_agent_check.py

```diff
@@ -157,6 +157,17 @@ def test_unicode_string(self):
         assert normalized_tags is not tags
         assert normalized_tag == tag.encode('utf-8')
 
+    def test_unicode_device_name(self):
+        check = AgentCheck()
+        tags = ['test:tag']
+        device_name = u'unicode_string'
+
+        normalized_tags = check._normalize_tags(tags, device_name)
+        normalized_device_tag = normalized_tags[1]
+
+        assert len(normalized_tags) is 2
+        assert isinstance(normalized_device_tag, str)
```

Comment:
The difference between python 2 and 3 in your example is mostly that `str` in python 2 can be either text or data, whereas `str` in python 3 is text data. So for py2/3 compatibility we actually want `_normalize_tags` to either always return binary data (`bytes` in both py2 and 3), or always return text data (`unicode` in py2, `str` in py3)

Also, unlike what's documented inline, I think we actually want `_normalize_tags` to return a list of `bytes` instead of a list of text data (see what `_normalize_tags_type` does).

So AFAIK `_normalize_tags` should also encode the `device:` tag to `bytes` instead of keeping it as text data. @ofek can confirm what the desired behavior is here.

Reply:
Yeah we want to have every tag in `bytes`, `utf-8` encoded. Appending the `device` tag should be done before calling the `_normalize_tags_type` method.

Reply:
Thanks for the comments here! I updated the `_normalize_tags()` function to add the device tag before calling `_normalize_tags_type()`